### DISCUSSION TITLE
How is Wind veer modelled?

### DISCUSSION THREAD

**Discussion by prith-gs** [NatLabRockies/floris#998] [General]
Hello, 

Can someone explain how wind veer is modelled in FLORIS?

The wind_veer parameter in the input file, how is it used?

I couldn't find much online. 
If there's a documentation on how wind veer is modelled in FLORIS that I can go through that'd be nice. 

Thanks!

**Comment by misi9170** [comment]
Hi @prith-gs ,

Veer (or any variation in wind direction) is not currently modeled ubiquitously within FLORIS; rather, it is a feature in just the GCH wake model. In particular, you’ll see that veer is used in the [gauss velocity model](https://github.com/NREL/floris/blob/main/floris/core/wake_velocity/gauss.py) and the [gauss deflection model](https://github.com/NREL/floris/blob/main/floris/core/wake_deflection/gauss.py), but not in other models. So, defining a value for veer while cause changes if using the GCH model, but not otherwise.

Best,
Misha

**Comment by prith-gs** [comment]
Hello, 

I have a follow up question. 

When I give the wind_veer = 10 degrees and wind direction = 90 degrees: What does that mean?

The wind direction is 10 degrees off with respect to what?

Thanks!

**Comment by misi9170** [comment]
Hi @prith-gs ,

The 10 degrees is taken as relative to the main wind direction; in the model, they are essentially independent. That is, regardless of the main wind direction, the veer represents the change in wind direction across the rotor (which is only used to construct the wake shape in the GCH model, as described above).

The exact form of this is given in Eq. (10) of [Annoni et al. (2018)](https://wes.copernicus.org/articles/3/819/2018/). 

I'm not sure if I've answered your question here, so feel free to clarify if I'm missing something!

Cheers,
Misha

**Comment by prith-gs** [comment]
Hi Misha, 

Thanks for your response. 

I have a simple test case with 6 Turbines spaced 7D apart. I ran it for series of wind veer values from -20 to 20 degrees. I was interested to see the change in power for the Turbines in wake.
![image](https://github.com/user-attachments/assets/b1eec095-e11d-488d-a9ef-8ba8ee01cdfb)

I noticed that a wind veer of 10 and -10 degrees is giving me the same numbers for power when there's no yaw offset which is why I came back to this discussion. But this makes sense when looking at equation 10.

![image](https://github.com/user-attachments/assets/2783f0b7-4dfe-4691-9fc9-66d1c7c283b5)

From NREL FLORIS libraries
![image](https://github.com/user-attachments/assets/da40f5ca-92ac-4bb6-b6b6-8586ac671961)

Equation 8 from the paper
![image](https://github.com/user-attachments/assets/654cb473-32ce-454d-958e-548fba6c7b8a)

'a' has cos^2 and sin^2, so a negative wind veer will have the same value as +ve veer
'b' has only sin, but for no yaw offset the equation won't change. This can be seen in equation 8 in the paper (posted above). sigma_y and sigma_z is same. So a negative wind veer will have the same value as +ve veer.
'c' again same as 'a' has cos^2 and sin^2, so a negative wind veer will have the same value as +ve veer

But a negative wind veer and positve wind veer should behave differently, right?

I'd like to address a previous discussion (#991) I raised where it was concluded that a turbine even with no yaw offset has a little bit of deflection which is because of the ground effects, shear etc. If this is the case, then how can negative veer and positive veer give me the same power values?

Looking forward to your response.

Best,
Pritheesh

**Reply by misi9170** [reply]
Hi Pritheesh,

Just to make sure I'm understanding correctly, are you saying that when you have the the `enable_secondary_steering` flag set to `True`, the downstream turbine produces precisely the same power for a veer of 10 and of -10, even though you are seeing a slight deflection of the wake as discussed in #991? 

That aside, to your question:
> But a negative wind veer and positive wind veer should behave differently, right?

I'll have to admit that I'm no expert in fluid interactions in the ABL. I'll look around to see if I can find someone that can give you a thorough response on that front

Misha
